### PR TITLE
[REF] Extract getSearchSQL from getSearchQuery.

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Code cleanup - function extraction

Before
----------------------------------------
Less readable

After
----------------------------------------
More readable

Technical Details
----------------------------------------
I made 2 changes over & above the function extraction

1) I changed one place in the code to call the new version - mostly for show - will do more later
2) I made it so the re-enable is called regardless of whether it was disabled after running the query. My logic is that it will only re-enabled IF it is enabled for the site so we should be safe. Doing this simplifies the code

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contact/BAO/Query.php [hunk 1]
```diff
@@ -4972,9 +4884,8 @@ public function searchQuery(
 
     $dao = CRM_Core_DAO::executeQuery($query);
 
-    if ($disableFullGroupByMode) {
-      CRM_Core_DAO::reenableFullGroupByMode();
-    }
+    // We can always call this - it will only re-enable if it was originally enabled.
+    CRM_Core_DAO::reenableFullGroupByMode();
```
Comment (eileenmcnaughton) [inline CRM/Contact/BAO/Query.php]: @monishdeb are you happy with this change based on the handling in the re-enable function being dependent on it having been enabled in the first place?

**Comment by colemanw** [comment]
@eileenmcnaughton is this covered by tests?

**Comment by eileenmcnaughton** [comment]
@colemanw yep - this function is heavily tested

**Comment by colemanw** [comment]
Merging based on code review and passing tests.

**Pull request merged**